src/package_name_validations.rs

```diff
@@ -0,0 +1,116 @@
+// Based on: https://github.com/rust-lang/cargo/blob/e975158c1b542aa3833fd8584746538c17a6ae55/src/cargo/ops/cargo_new.rs#L169
+pub fn cargo_check_name(name: &str) -> anyhow::Result<()> {
+    // Instead of `PackageName::new` which performs these checks in the original cargo code
+    validate_package_name(name)?;
+
+    if is_keyword(name) {
+        anyhow::bail!(
+            "the name `{}` cannot be used as a package name, it is a Rust keyword",
+            name,
+        );
+    }
+    if is_conflicting_artifact_name(name) {
+        anyhow::bail!(
+            "the name `{}` cannot be used as a package name, \
+                it conflicts with cargo's build directory names",
+            name,
+        );
+    }
+    if name == "test" {
+        anyhow::bail!(
+            "the name `test` cannot be used as a package name, \
+            it conflicts with Rust's built-in test library",
+        );
+    }
+    if ["core", "std", "alloc", "proc_macro", "proc-macro"].contains(&name) {
+        eprintln!(
+            "⚠️  Warning: the name `{}` is part of Rust's standard library\n\
+            It is recommended to use a different name to avoid problems.",
+            name,
+        );
+    }
+    if is_windows_reserved(name) {
```

Comment:
The original code bailed if the platform is actually windows, I think this should be good enough for now